core_lib/src/managers/selectionmanager.cpp

```diff
@@ -0,0 +1,377 @@
+    mSelection(QRectF()),
+    mTempTransformedSelection(QRectF()),
+    mTransformedSelection(QRectF()),
+    mRotatedAngle(0),
+    mOldRotatedAngle(0),
+    mSomethingSelected(false),
+    mLastSelectionPolygonF(QPolygonF()),
+    mCurrentSelectionPolygonF(QPolygonF()),
+    mOffset(QPointF()),
+    mMoveMode(MoveMode::NONE)
+{
+}
+
+SelectionManager::~SelectionManager()
+{
+}
+
+bool SelectionManager::init()
+{
+    return true;
+}
+
+Status SelectionManager::load(Object*)
+{
+    return Status::OK;
+}
+
+Status SelectionManager::save(Object*)
+{
+    return Status::OK;
+}
+
+void SelectionManager::workingLayerChanged(Layer *)
+{
+}
+
+void SelectionManager::resetSelectionTransformProperties()
+{
+    mOffset = QPointF(0, 0);
+    mRotatedAngle = 0;
+    mOldRotatedAngle = 0;
+    mSelectionTransform.reset();
+}
+
+void SelectionManager::updatePolygons()
+{
+    mCurrentSelectionPolygonF = mTempTransformedSelection;
+    mLastSelectionPolygonF = mTransformedSelection;
+}
+
+void SelectionManager::resetSelectionTransform()
+{
+    mSelectionTransform.reset();
+}
+
+bool SelectionManager::isOutsideSelectionArea(QPointF point)
+{
+    return (!mTransformedSelection.contains(point)
+            && validateMoveMode(point) == MoveMode::NONE);
+}
+
+void SelectionManager::deleteSelection()
+{
+    emit needDeleteSelection();
+}
+
+void SelectionManager::clearCurves()
+{
+    mClosestCurves.clear();
+}
+
+void SelectionManager::clearVertices()
+{
+    mClosestVertices.clear();
+}
+
+qreal SelectionManager::selectionTolerance() const
+{
+    return qAbs(mSelectionTolerance * editor()->viewScaleInversed());
+}
+
+MoveMode SelectionManager::validateMoveMode(QPointF pos)
+{
+    return moveModeForAnchorInRange(pos);
+}
+
+MoveMode SelectionManager::moveModeForAnchorInRange(QPointF lastPos)
+{
+    QRectF transformRect = mTempTransformedSelection;
+    QPointF lastPoint = lastPos;
+    const qreal& selectionTolerance = mSelectionTolerance;
+
+    MoveMode mode;
+    if (QLineF(lastPoint, transformRect.topLeft()).length() < selectionTolerance)
+    {
+        mode = MoveMode::TOPLEFT;
+    }
+    else if (QLineF(lastPoint, transformRect.topRight()).length() < selectionTolerance)
+    {
+        mode = MoveMode::TOPRIGHT;
+    }
+    else if (QLineF(lastPoint, transformRect.bottomLeft()).length() < selectionTolerance)
+    {
+        mode = MoveMode::BOTTOMLEFT;
+
+    }
+    else if (QLineF(lastPoint, transformRect.bottomRight()).length() < selectionTolerance)
+    {
+        mode = MoveMode::BOTTOMRIGHT;
+    }
+    else if (mTransformedSelection.contains(lastPoint))
+    {
+        mode = MoveMode::MIDDLE;
+    }
+    else {
+        mode = MoveMode::NONE;
+    }
+    mMoveMode = mode;
+    return mode;
+}
+
+MoveMode SelectionManager::getMoveModeForSelectionAnchor(QPointF pos)
+{
+    const double radius = mSelectionTolerance / 2;
+    const qreal selectionTolerance = mSelectionTolerance;
```

Comment:
Now we're not taking the zoom into consideration at all. This isn't exactly what I mean. I think we should still use the view and I like that you've used one variable for all of the selection tolerances, but I think it should come from the function `selectionTolerance()` to compensate for the view.

Also it looks like `mSelectionTolerance` *is* the radius, so no need to divide by 2. Look at `moveModeForAnchorInRange` for example.

Reply:
Whoops.. completely went over my head that I shouldn't use mSelectionTolerance 🤦‍♂